### PR TITLE
feat(website/search): create search engine

### PR DESCRIPTION
**Pull request by gael-boyenval**
## I have read [the contributing guidelines](https://mozaic.adeo.cloud/Contributing/)

- [x] Yes
- [ ] No

## Does this PR introduce a breaking change?

read [What is a breaking change ?](https://mozaic.adeo.cloud/Contributing/Prerequisite/GitConventions/#breaking-changes-)

- [ ] Yes
- [x] No

## Describe the changes

Github issue number or Jira issue URL: N/A

create a search engine for the documentation

### CODE REVIEW

**Comment by nherwegh** [comment]
I'd rather say that the first version was the better. For example, I searched "layout" on both solutions. The first solution gave me a good answer right away. The other one, with Aglolia didn't give me the answer I was looking for. Maybe try to do the same to see what I'm talking about

**Comment by gael-boyenval** [comment]
@nherwegh : done

File: packages/gatsby-theme-styleguide/src/gatsby-components/Search/Search.js [hunk 1]
```diff
@@ -0,0 +1,198 @@
...
+  font-size: 1rem;
+  text-transform: none;
+  box-shadow: 0 0.3rem 0.3rem 0 rgba(0, 0, 0, 0.3);
+  border-radius: 6px;
+  overflow: hidden;
+`
+
+const SearchWrapper = styled.div`
+  position: fixed;
+  right: 0.75rem;
```
Comment (tiloyi) [inline packages/gatsby-theme-styleguide/src/gatsby-components/Search/Search.js]: 1.875rem

File: packages/gatsby-theme-styleguide/src/gatsby-components/Search/Search.js [hunk 2]
```diff
@@ -0,0 +1,198 @@
...
+  border-radius: 4px;
+}
+
+&::placeholder {
+  font-size: 0.875rem;
+  line-height: 1.5714286;
+}
+
+&.is-focus,
+&:focus {
```
Comment (tiloyi) [inline packages/gatsby-theme-styleguide/src/gatsby-components/Search/Search.js]: **&.is-focus** => not necessary

File: packages/gatsby-theme-styleguide/src/gatsby-components/Search/Search.js [hunk 3]
```diff
@@ -0,0 +1,198 @@
...
+        <ResulWrapper>
+          <ResultList />
+        </ResulWrapper>
+      </SearchWrapper>
+    )
+  }
+
+  getSearchResults(query) {
+    // adicionar variável para língua
+    var index = window.__FLEXSEARCH__.en.index
```
Comment (mohamedMok) [inline packages/gatsby-theme-styleguide/src/gatsby-components/Search/Search.js]: Maybe use English comments!

**Pull request merged**